www/content/tutorial.md

```diff
@@ -1577,373 +1603,324 @@ Including the hash solves a number of problems:
-
 ### [Platform Modules](#platform-modules) {#platform-modules}
 
 \[This part of the tutorial has not been written yet. Coming soon!\]
 
-See [Platform Switching Rust](https://github.com/roc-lang/roc/blob/main/examples/platform-switching/rust-platform/main.roc) for an example.
-
 ### [Importing Files](#importing-files) {#importing-files}
 
 You can import files directly into your module as a `Str` or a `List U8` at compile time. This is can be useful when working with data you would like to keep in a separate file, e.g. JSON or YAML configuration.
 
 ```roc
-import "some-file" as someStr : Str
-import "some-file" as someBytes : List U8
+import "some-file" as some_str : Str
+import "some-file" as some_bytes : List U8
 ```
 
 See the [Ingest Files Example](https://www.roc-lang.org/examples/IngestFiles/README.html) for a demonstration on using this feature.
 
-## [Tasks](#tasks) {#tasks}
+## [Effectful functions](#efffectful-functions) {#efffectful-functions}
+
+There are two types of functions in roc, "pure" and "effectful". Consider these two functions:
 
-Note: [Tasks will soon dissolve entirely for app developers](#purity-inference), but today they're still critical.
+```roc
+with_extension : Str -> Str
+with_extension = \filename ->
+    "$(filename).roc"
 
-Tasks are provided in a builtin `Task` module like the `List`, `Str` modules. They're an important part of building Roc applications, so let's continue using the [basic-cli](https://github.com/roc-lang/basic-cli) platform we've been using up to this point as an example!
+read_file! : Str => Str
+read_file! = \path ->
+    File.read_utf8! with_extension(path) |> Result.withDefault ""
+```
 
-In the `basic-cli` platform, here are four operations we can do:
+Notice the subtle difference in these functions' types:
+
+- Str `->` Str is the type of the pure function `with_extension`
+- Str `=>` Str is the type of the effectful function `read_file!`
+
+The arrow in the type tells you whether a function is pure or might perform effects. Pure functions use `->` and effectful functions use `=>`. Effectful functions can call either pure functions or other effectful functions, but pure functions can only call other pure functions. (The fact that pure functions can only call pure functions is part of the definition of pure functions; Roc is just reflecting that fact in the type system.)
+
+The `!` is a suffix character inside function names themselves, marking them as effectful. This is just a naming convention, though it is enforced by the compiler with a warning.
+
+Note: this was a previous change to the language, and `!` used to be an operator, so you may still find code that as a `Task` in it or an effectful function like `main` without a `!` suffix. This code should be upgraded to use `Result` and the naming convention for purity inference, e.g. `main!`. For more information about this change, see [this recent talk](https://www.youtube.com/watch?v=42TUAKhzlRI).
```

Comment:
```suggestion
Note: this was a previous change to the language, and `!` used to be an operator, so you may still find code that has a `Task` in it or an effectful function like `main` without a `!` suffix. This code should be upgraded to use `Result` and the naming convention for purity inference, e.g. `main!`. For more information about this change, see [this recent talk](https://www.youtube.com/watch?v=42TUAKhzlRI).
```

Also:
* I don't fully comprehend what this fragment "this was a previous change to the language" is about. But it may be just me.
* The "recent talk" may shortly become not-so-recent. Perhaps the word "recent" can be removed.

Reply:
Done in 7cd9fae334